### DISCUSSION TITLE
Custom mg_set_non_blocking_mode() implementation

### DISCUSSION THREAD

**Discussion by dmitry-kandiner** [cesanta/mongoose#1410] [Feature requests]
It would be nice to have an option for custom **mg_set_non_blocking_mode()** implementation - some architectures have non-standard way for doing so. For instance, Cyclone TCP stack implementation of **fcntl()** requires three parameters: a socket, a command and a pointer to the command parameters.

**Comment by dmitry-kandiner** [comment]
Resolved here: https://github.com/cesanta/mongoose/issues/1411

**Comment by cpq** [comment]
For the sake of record. SDK's version of `fcntl()` differs from the standard, and therefore the solution is to have a snippet in `mongoose_custom.h` to replace `fcntl`:

```c
#include <core/bsd_socket.h>

#define F_SETFD 12345
#define FD_CLOEXEC 0

static inline int mg_fcntl(int fd, int cmd, ...) {
  if (cmd == F_GETFL) { uint32_t opt; fcntl(fd, cmd, &opt); return opt; }
  ...
}

#define fcntl mg_fcntl
```
